fenix/app/src/main/java/org/mozilla/fenix/onboarding/OnboardingFragment.kt

```diff
@@ -0,0 +1,248 @@
+import org.mozilla.fenix.search.toolbar.SearchSelectorMenu
+import java.lang.ref.WeakReference
+
+/**
+ * Displays the first run onboarding screen.
+ */
+class OnboardingFragment : Fragment() {
+
+    private var _binding: FragmentHomeBinding? = null
+    private val binding get() = _binding!!
+
+    private val searchSelectorMenu by lazy {
+        SearchSelectorMenu(
+            context = requireContext(),
+            interactor = interactor,
+        )
+    }
+
+    private val store: BrowserStore
+        get() = requireComponents.core.store
+    private val browsingModeManager
+        get() = (activity as HomeActivity).browsingModeManager
+
+    private var _interactor: DefaultOnboardingInteractor? = null
+    private val interactor: DefaultOnboardingInteractor
+        get() = _interactor!!
+
+    private var onboardingView: OnboardingView? = null
+    private var homeMenuView: HomeMenuView? = null
+    private var tabCounterView: TabCounterView? = null
+    private var toolbarView: ToolbarView? = null
+
+    private lateinit var onboardingStore: OnboardingStore
+    private lateinit var onboardingAccountObserver: OnboardingAccountObserver
+
+    private val searchSelectorBinding = ViewBoundFeatureWrapper<SearchSelectorBinding>()
+    private val searchSelectorMenuBinding = ViewBoundFeatureWrapper<SearchSelectorMenuBinding>()
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?,
+    ): View {
+        _binding = FragmentHomeBinding.inflate(inflater, container, false)
+
+        val activity = activity as HomeActivity
+
+        onboardingAccountObserver = OnboardingAccountObserver(
+            context = requireContext(),
+            dispatchChanges = ::dispatchOnboardingStateChanges,
+        )
+
+        onboardingStore = StoreProvider.get(this) {
+            OnboardingStore(
+                initialState = OnboardingFragmentState(
+                    onboardingState = onboardingAccountObserver.getOnboardingState(),
+                ),
+            )
+        }
+
+        _interactor = DefaultOnboardingInteractor(
+            controller = DefaultOnboardingController(
+                activity = activity,
+                navController = findNavController(),
+                hideOnboarding = ::hideOnboarding,
+            ),
+            privateBrowsingController = DefaultPrivateBrowsingController(
+                activity = activity,
+                appStore = requireComponents.appStore,
+                navController = findNavController(),
+            ),
+            searchSelectorController = DefaultSearchSelectorController(
+                activity = activity,
+                navController = findNavController(),
+            ),
+            toolbarController = DefaultToolbarController(
+                activity = activity,
+                store = requireComponents.core.store,
```

Comment:
this can just be your local variable `store`

Reply:
True, gotta make use of those getters